test/unit/spec/EnrollSms_spec.js

```diff
@@ -262,59 +298,68 @@ function (Okta, OktaAuth, LoginUtil, Util, AuthContainer, Form, Beacon, Expect,
             });
           });
       });
+
       itp('clears previous errors in form when resend code', function () {
-        return setupAndSendInvalidCode().then(function (test) {
-          expect(test.form.hasErrors()).toBe(true);
-          expect(test.form.errorMessage()).toBe('Invalid Phone Number.');
-          return sendCodeOnEnter(test, successRes, 'US', '4151111111');
-        })
+        return setupAndSendInvalidCode()
+          .then(function (test) {
+            expect(test.form.hasErrors()).toBe(true);
+            expect(test.form.errorMessage()).toBe('Invalid Phone Number.');
+            sendCodeOnEnter(test, successRes, 'US', '4151111111');
+            return waitForEnrollActivateSuccess(test);
+          })
           .then(function (test) {
             expect(test.form.hasErrors()).toBe(false);
           });
       });
+
       itp('validation error if phone number field is blank', function () {
         return sendCodeFn(successRes, 'US', '')
           .then(function (test) {
             expect(test.form.hasErrors()).toBe(true);
           });
       });
 
-      itp('shows warning message to click "Re-send" after 30s', function () {  
-        Util.speedUpDelay();      
-        return setup().then(function (test) {
-          test.setNextResponse(resFactorEnrollActivateSms);
-          enterCode(test, 'US', '4151111111');
-          test.form.sendCodeButton().click();
-          return tick(test);
-        })
+      itp('shows warning message to click "Re-send" after 30s', function () {
+        Util.speedUpDelay();
+        return sendCodeFn(successRes, 'US', '4151111111')
+          .then(waitForEnrollActivateSuccess)
```

Comment:
isn't the `sendCodeFn` already waiting for enrollActivateSuccess? Similar to what you are doing in Call tests, the changes to the sendCode functions have an `Expect.watifForEnrollActivateSuccess` now... or maybe this is a different sendCodeFn?

Reply:
Not really (too much sharing cause confusing sometime)
`sendCodeFn` for `testEnrollPhoneNumber` is either `setupAndSendCode` or `setupAndSendCodeIdx` which only does `setup` and `sendCode`.
`setupAndSendValidCode` does set up `setup`, `sendCode` and `waitForSuccess`

This test may deserve further refactors to make it more clear but I gave up at this PR to reduce the scope.